src/gameState.ts

```diff
@@ -0,0 +1,299 @@
+
+    public display():void{
+        console.log();
+        console.log(this.board);
+        console.log();
+    }
+
+    public isValidTarget(position:Position):boolean{
+        if (position.row >= 0){
+            if (position.row < 10){
+                if (position.column >= 0){
+                    if (position.column < 10){
+                        return (this.board[position.row][position.column]==0);
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    public eliminateSunkenShips():void{
+        // first scans rows for sunken ships
+        for (let row:number =0; row<10; row++){
+            for (let column:number = 0; column<10; column++){
+                let startColumn:number = column;
+                if (this.board[row][column]==1){
+                    while ((column<10) && (this.board[row][column]==1)){
+                        column++;
+                    }
+                    let sunken:boolean = this.isSunken(row,startColumn,row,column-1);
+                    if (sunken){
+                        let shipLength:number = column-startColumn;
+                        this.remainingShips = this.remove(shipLength);
+                        this.sink(row,startColumn,row,column-1);
+                    }
+                }
+            }
+        }
+        // then checks columns for sunken ships
+        for (let column:number =0; column<10; column++){
+            for (let row:number = 0; row<10; row++){
+                let startRow:number = row;
+                if (this.board[row][column]==1){
+                    while ((row<10) && (this.board[row][column]==1)){
+                        row++;
+                    }
+                    let sunken:boolean = this.isSunken(startRow,column,row-1,column);
+                    if (sunken){
+                        let shipLength:number = row-startRow;
+                        this.remainingShips = this.remove(shipLength);
+                        this.sink(startRow,column,row-1,column);
+                    }
+                }
+            }
+        }
+    }
+
+    public remove(shipLength:number):number[]{
+        let ans:number[] = [];
+        let done:boolean = false;
+        for (let i:number = 0; i<this.remainingShips.length; i++){
+            if (done){
+                ans.push(this.remainingShips[i]);
+            }
+            else {
+                if (this.remainingShips[i]!=shipLength){
+                    ans.push(this.remainingShips[i]);
+                }
+                else{
+                    done = true;
+                }
+            }
+        }
+        return ans;
+    }
+
+    public eliminateNeighboursOfSunken():void{
+        let x:number[] = [1,1,1,0,-1,-1,-1,0];
+        let y:number[] = [-1,0,1,1,1,0,-1,-1];
+        for (let row:number = 0; row < 10; row++){
+            for (let column:number = 0; column < 10; column ++){
+                if (this.board[row][column] == 2){
```

Comment:
Again, I have no idea what a 2 means here on a brief reading. As above, it could be extracted into a class that carried your row/column matrix, and had named methods for what 0, 1 & 2 mean!